client/src/parsec/classes.ts

```diff
@@ -0,0 +1,469 @@
+            email: 'gale@waterdeep.faerun',
+            // cspell:disable-next-line
+            label: 'Gale Dekarios',
+          },
+        }});
+      });
+    }
+  }
+
+  async initialWaitHost():
+  Promise<Result<DeviceClaimInProgress1Info, ClaimInProgressError>> {
+    this._assertState(true, false);
+    if (window.isDesktop()) {
+      this.canceller = await libparsec.newCanceller();
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const result = await libparsec.claimerDeviceInitialDoWaitPeer(this.canceller, this.handle!);
+      if (result.ok) {
+        this.SASCodeChoices = result.value.greeterSasChoices;
+        this.correctSASCode = result.value.greeterSas;
+        this.handle = result.value.handle;
+      }
+      this.canceller = null;
+      return result;
+    } else {
+      this.SASCodeChoices = ['5MNO', '6PQR', '7STU', '8VWX'];
+      this.correctSASCode = '7STU';
+      return new Promise<Result<DeviceClaimInProgress1Info, ClaimInProgressError>>((resolve, _reject) => {
+        resolve({ok: true, value: {
+          handle: DEFAULT_HANDLE,
+          greeterSas: '7STU',
+          greeterSasChoices: ['5MNO', '6PQR', '7STU', '8VWX'],
+        }});
+      });
+    }
+  }
+
+  async signifyTrust():
+  Promise<Result<DeviceClaimInProgress2Info, ClaimInProgressError>> {
+    this._assertState(true, false);
+    if (window.isDesktop()) {
+      this.canceller = await libparsec.newCanceller();
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const result = await libparsec.claimerDeviceInProgress1DoSignifyTrust(this.canceller, this.handle!);
+      if (result.ok) {
+        this.guestSASCode = result.value.claimerSas;
+        this.handle = result.value.handle;
+      }
+      this.canceller = null;
+      return result;
+    } else {
+      this.guestSASCode = '1337';
+      return new Promise<Result<DeviceClaimInProgress2Info, ClaimInProgressError>>((resolve, _reject) => {
+        resolve({ok: true, value: {
+          handle: DEFAULT_HANDLE,
+          claimerSas: '1337',
+        }});
+      });
+    }
+  }
+
+  async waitHostTrust():
+  Promise<Result<DeviceClaimInProgress3Info, ClaimInProgressError>> {
+    this._assertState(true, false);
+    if (window.isDesktop()) {
+      this.canceller = await libparsec.newCanceller();
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const result = await libparsec.claimerDeviceInProgress2DoWaitPeerTrust(this.canceller, this.handle!);
+      this.canceller = null;
+      if (result.ok) {
+        this.handle = result.value.handle;
+      }
+      return result;
+    } else {
+      await _wait(this.MOCK_HOST_WAITING_TIME);
+      return new Promise<Result<DeviceClaimInProgress3Info, ClaimInProgressError>>((resolve, _reject) => {
+        resolve({ok: true, value: {
+          handle: DEFAULT_HANDLE,
+        }});
+      });
+    }
+  }
+
+  async doClaim(deviceLabel: string):
```

Comment:
It makes the API a little clearer to type it as `DeviceLabel` (even though it is just a string underneath).